internal/app/machined/pkg/runtime/v1alpha1/platform/metal/metal.go

```diff
@@ -29,6 +29,19 @@ const (
 	mnt = "/mnt"
 )
 
+type noConfigSource struct{}
+
+func (ncs *noConfigSource) Error() string {
+	return "no configuration source"
+}
+
+func (ncs *noConfigSource) NoConfigurationSource() {
+}
+
+func errNoConfigSource() error {
+	return new(noConfigSource)
+}
```

Comment:
I noticed this is also defined in `container.go`. Maybe we could find a common place?

Reply:
Yeah, a common place would definitely be better.  Unfortunately, `platform` is not it, because platform imports each of these, thus creating an import loop.

Reply:
Ah, well :hankey: . :thinking: what if we had `platform/errors`?

Reply:
Works for me.